src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncExceptionHandlerRewriter.cs

```diff
@@ -458,7 +459,8 @@ private BoundStatement RewriteFinalizedRegion(BoundTryStatement node)
             var origAwaitCatchFrame = _currentAwaitCatchFrame;
             _currentAwaitCatchFrame = null;
 
-            var rewrittenCatches = this.VisitList(node.CatchBlocks);
+            var rewrittenCatches = node.CatchBlocks.SelectAsArray(catchBlock =>
```

Comment:
>SelectAsArray [](http://example.com/codeflow?start=52&length=13)

Is there an overload that takes another argument and passes it to the delegate. If not, perhaps we should add one and use it in order to avoid capturing ```this``` and ```origAwaitCatchFrame``` in a closure.